src/brsTypes/components/RoSGNode.ts

```diff
@@ -69,27 +71,21 @@ export class RoSGNode extends BrsComponent implements BrsValue, BrsIterable {
     private fields = new Map<string, Field>();
     private children: RoSGNode[] = [];
     private parent: RoSGNode | BrsInvalid = BrsInvalid.Instance;
-    readonly builtInFields = [
+    readonly defaultFields: FieldModel[] = [
         { name: "change", type: "roAssociativeArray" },
         { name: "focusable", type: "boolean" },
         { name: "focusedChild", type: "node" },
         { name: "id", type: "string" },
     ];
 
-    constructor(members: AAMember[], readonly type: string = "Node") {
-        super("roSGNode");
+    constructor(initializedFields: AAMember[], readonly type: string = "Node") {
+        super(type);
 
-        // All nodes start have some built-in fields when created
-        this.builtInFields.forEach(field => {
-            this.fields.set(
-                field.name.toLowerCase(),
-                new Field(getBrsValueFromFieldType(field.type), false)
-            );
-        });
+        // All nodes start have some built-in fields when created.
+        this.registerDefaultFields(this.defaultFields);
 
-        members.forEach(member =>
-            this.fields.set(member.name.value.toLowerCase(), new Field(member.value, false))
-        );
+        // After registering default fields, then register fields instantiated with initial values.
+        this.registerInitializedFields(initializedFields);
```

Comment:
I renamed this to `initializedFields` to make the relationship between these and `defaultFields` more clear.